### DISCUSSION TITLE
Member type for IFS

### DISCUSSION THREAD

**Discussion by SteversJE** [codefori/vscode-ibmi#2306] [Q&A] [answered]
How do I get the source member type into an action for use in copying to the IFS?

I want to take RPG or RPGLE or SQLRPGLE from here...

![image](https://github.com/user-attachments/assets/ede0603e-06c6-4807-9b1a-0b21b36abb99)

... through some action like this (everything works but the member type)...

`CPYTOSTMF FROMMBR('/QSYS.LIB/&OPENLIB.LIB/&OPENSPF.FILE/&OPENMBR.MBR') TOSTMF('/QIBM/UserData/source/&OPENLIB/&OPENSPF/&OPENMBR.&MBR')`

and end up here with the proper name.

![image](https://github.com/user-attachments/assets/c573b661-0452-4ce9-94a7-1c5184453449)

I could not find any reference document that helps.

**Comment by sebjulliand** [comment, marked as answer]
Try using `&EXT` to get the member's type (or `&EXTL` for lowercase).
This is described in the documentation displayed when you create/edit an action:
![image](https://github.com/user-attachments/assets/c82fb631-148d-400a-a632-f03bef1a9141)

**Reply by SteversJE** [reply]
Thanks. I haven't got to that screen yet as I was editing the command in the workspace JSON file and searching for online documentation.

**Comment by SteversJE** [comment]
.
